### ISSUE TITLE
You can sell ships on other planets

### ISSUE THREAD

**Issue by TheGiraffe3** [endless-sky/endless-sky#10801]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I am flying a Wasp, and land on a planet with a shipyard. I have a Palavret on a different planet in the same system.
If I sell all ships I own on my current planet, I can sell the Palavret, even though it's not possible for me to get to it.

### Steps to Reproduce

1. Be on a planet with a shipyard.
2. Another planet in the same system has parked ships you own.
3. Sell all your ships on the current planet.
4. Observe that you can sell your ships on the other planet.

### Expected Behavior

You can't sell ships you can't get to.

### Screenshots

![Screenshot 2024-11-20 at 4 24 07 PM](https://github.com/user-attachments/assets/921e5960-ef3f-4900-b607-fdf23262e2d9)
![Screenshot 2024-11-20 at 4 24 11 PM](https://github.com/user-attachments/assets/164e24db-f768-4c0f-a9c7-951128b7f955)
![Screenshot 2024-11-20 at 4 24 14 PM](https://github.com/user-attachments/assets/24c20c3e-6092-4670-8f42-6f9b343ece0d)
![Screenshot 2024-11-20 at 4 24 16 PM](https://github.com/user-attachments/assets/e1137458-501f-4513-b119-e0a7cbee1381)
![Screenshot 2024-11-20 at 4 24 45 PM](https://github.com/user-attachments/assets/55aa75c3-01f3-4e1b-b4bd-06fcecca3840)

Ignore the Jibauni ships shown.

### Link to save file

I did find this when testing the [Jibauni](https://github.com/TheGiraffe3/Jibauni), so the plugin may have affected it, but I don't think that would change anything.
[Jibauni Tester.txt](https://github.com/user-attachments/files/17831218/Jibauni.Tester.txt)

### Operating System

macOS Ventura 13.7.1

### Game Source

GitHub Releases

### Game Version

0.10.11 continuous

### Additional Information

The other planet also has a shipyard, which could affect the sale.
My guess is that when selling ships the shipyard only checks for the system the ship is in; not the planet, and that's the problem.
Don't know exactly how to fix that though.

**Comment by TheGiraffe3** [comment]
Upon further delving in, it looks like this is the problem:
https://github.com/endless-sky/endless-sky/blob/839f5af27fa92c7030284502648e81a1baad0e31/source/ShipyardPanel.cpp#L389
I believe changing `GetSystem` to `GetPlanet` would fix this?

**Comment by xobes** [comment]
I can take this one.

**Comment by Amazinite** [comment]
There's another issue somewhere on this. I believe it's intentional.

**Comment by Amazinite** [comment]
#5210

**Comment by xobes** [comment]
I have confirmed that the line identified will affect this behavior as TheGiraffe3 identified.

An interested note related to this is that while ships indeed are parked on specific planets they are not displayed in the Player Info as being on a planet (only in a system).

**Comment by xobes** [comment]
Actually, I'm not able to reproduce the problem on the master branch...

I see parked ships only on the planet I was on when I parked them.  Oh.... You have to sell everything first.

Actually, it's weirder than that, you can sell "most" and if you sell past the index of that parked ship (maybe) then you see the parked ship get selected.

![Screenshot from 2024-11-21 18-30-26](https://github.com/user-attachments/assets/1235f20d-926f-446e-8481-6cd9ad280d95)

Anyway if the idea is to not fix this due to "frustration" (https://github.com/endless-sky/endless-sky/issues/5210#issuecomment-669412856) then I would think that you'd either want ships to be parked in a System or else the Player Info panel identify the Planet (not just the System) that they are parked on.

**Comment by xobes** [comment]
I have confirmed the behavior described.
And I have confirmed that fix identified by TheGiraffe3 does fix the problem.

**Comment by petervdmeer** [comment]
> There's another issue somewhere on this. I believe it's intentional.

Yes, it was intentional, but we might want to consider changing this, since it apparently does not only provide "quality of life", but also generates some confusion.

**Comment by TheGiraffe3** [comment]
I think adding it to a help dialog or something would probably resolve this; without explanation, though, it seems like a bug.

**Comment by xobes** [comment]
I think that the real problem here is that ships are parked on planets but the player info panel only shows the system. 

And also, just fix this.

The whole argument that it's anti frustration is it the window given how parking works. 

The only time one will see that ship parked on a neighboring planet is when selling others.  It's not even helpful just broken logic.
